File: Exercism/ExercismFSharp/ExercismFSharp/Matrix.fs
module Matrix

open System

let fromString (input : string) =
    input.Split [|'\n'|]
    |> Array.map (fun row ->
        row.Split [|' '|]
        |> Array.map Int32.Parse
    )

let rows matrix =
    matrix

let cols matrix =
    let numRows = matrix |> Array.length
    let numCols = matrix |> Array.head |> Array.length

    [|0 .. numCols - 1|]
    |> Array.map (fun j ->
        [|0 .. numRows - 1|]
        |> Array.map (fun i ->
            matrix.[i].[j]
        )
    )

File: Exercism/python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):
        self.rows = [list(map(int, row.split())) for row in matrix_string.splitlines()]
        self.cols = [[self.rows[j][i] for j in range(0, len(self.rows))] for i in range(0, len(self.rows[0]))]

    def row(self, index):
        return list(self.rows[index - 1])

    def column(self, index):
        return list(self.cols[index - 1])
